webpush/encrypt.go

```diff
@@ -270,7 +281,7 @@ func hkdf(salt, ikm, info []byte, length int) []byte {
 	mac = hmac.New(sha256.New, prk)
 	mac.Write(info)
 	mac.Write([]byte{1})
-	return mac.Sum(nil)[0:length]
+	return mac.Sum(nil)[0:length], nil
```

Comment:
As currently written, `len(b)` is always 32, because it is a SHA256 hash. Your code is functionally equivalent to the current code, although a little more future proof.

Right now we have three options for what `hkdf` should do with a high length
1. panic
2. error (proposed behaviour)
3. silently return an incorrect result (current behaviour)

None of these things can **actually** happen right now because the lengths are hard coded at each call site. However, it came up in security review that it would be appropriate to code defensively, and 3 is the only actually dangerous option.